### PR TITLE
update display obj on union change

### PR DESCRIPTION
**Pull request by MattDHill**

### CODE REVIEW

File: frontend/projects/ui/src/app/components/form-object/form-object.component.ts [hunk 1]
```diff
@@ -63,6 +64,37 @@ export class FormObjectComponent {
...
+    // setTimeout hack to avoid ExpressionChangedAfterItHasBeenCheckedError
+    setTimeout(() => {
+      if (this.original) {
+        Object.keys(this.current || {}).forEach(key => {
+          if ((this.original as Config)[key] === undefined) {
+            this.hasNewOptions.emit()
+          }
+        })
+      }
+    }, 10)
```
Comment (waterplea) [inline frontend/projects/ui/src/app/components/form-object/form-object.component.ts:78]: You don't need any time, just default 0 is enough to move it to the next frame.

File: frontend/projects/ui/src/app/components/form-object/form-object.component.ts [hunk 2]
```diff
@@ -63,6 +64,37 @@ export class FormObjectComponent {
...
 
   ngOnInit() {
+    this.setDisplays()
+
+    // setTimeout hack to avoid ExpressionChangedAfterItHasBeenCheckedError
+    setTimeout(() => {
+      if (this.original) {
+        Object.keys(this.current || {}).forEach(key => {
+          if ((this.original as Config)[key] === undefined) {
+            this.hasNewOptions.emit()
```
Comment (waterplea) [inline frontend/projects/ui/src/app/components/form-object/form-object.component.ts:74]: Do we need it for every new option?

File: frontend/projects/ui/src/app/components/form-object/form-object.component.ts (hunk 1 above)
Comment (MattDHill) [reply frontend/projects/ui/src/app/components/form-object/form-object.component.ts:78]: fixed

File: frontend/projects/ui/src/app/components/form-object/form-object.component.ts (hunk 2 above)
Comment (MattDHill) [reply frontend/projects/ui/src/app/components/form-object/form-object.component.ts:74]: No, if any options are new, we can emit and abort

**Review by elvece — APPROVED** [review]

**Pull request merged**